apollo-rx2support/src/main/java/com/apollographql/android/rx2/Rx2Apollo.java

```diff
@@ -0,0 +1,148 @@
+package com.apollographql.android.rx2;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+import com.apollographql.apollo.ApolloWatcher;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.internal.util.Cancelable;
+
+import javax.annotation.Nonnull;
+
+import io.reactivex.Completable;
+import io.reactivex.CompletableEmitter;
+import io.reactivex.CompletableOnSubscribe;
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.Single;
+import io.reactivex.SingleEmitter;
+import io.reactivex.SingleOnSubscribe;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.exceptions.Exceptions;
+
+import static com.apollographql.apollo.api.internal.Utils.checkNotNull;
+
+/**
+ * The Rx2Apollo class provides methods for converting ApolloCall, ApolloPrefetch
+ * and ApolloWatcher types to RxJava 2 sources.
+ */
+public class Rx2Apollo {
+
+  private Rx2Apollo() {
+    throw new AssertionError("This class cannot be instantiated");
+  }
+
+  /**
+   * Converts an {@link ApolloPrefetch} to an asynchronous Observable.
```

Comment:
technically the Observable is not async, it has a callback but until someone uses a scheduler everything with still run synchronous by default